nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -122,7 +122,7 @@ sub GetFs {
         my @superOptions = split /,/, $fields[$n + 2];
 
         # Skip the read-only bind-mount on /nix/store.
-        next if $mountPoint eq "/nix/store" && (grep { $_ eq "rw" } @superOptions) && (grep { $_ eq "ro" } @mountOptions);
```

Comment:
they refer to the same device as another mountpoint which is a prefix.
My perl knowledge would be insufficient to implement this, though.